File: 2022/OpenCV4 tutorial/7 Image resample/resize_example.py
import cv2 as cv


def main():

    src = cv.imread("merge_split.jpg")

    """
        resize 常用模式：
            INTER_NEAREST 近邻插值
            INTER_LINEAR 线性插值
            INTER_CUBIC 三次插值
            INTER_AREA 
            INTER_LANCZOS4 三角插值
            INTER_LINEAR_EXACT
            INTER_NEAREST_EXACT
    """
    # 第2个参数为None的时候，第4(fx)、5(fy)个参数的两个浮点数，分别代表图片的宽放缩倍数、高放缩倍数
    dst = cv.resize(src, None, None, 3., 2., cv.INTER_NEAREST)
    cv.imshow("INTER_NEAREST", dst)
    dst = cv.resize(src, None, fx=3., fy=2., interpolation=cv.INTER_LINEAR)
    cv.imshow("INTER_LINEAR", dst)
    cv.waitKey()


if __name__ == '__main__':
    main()

File: 2022/OpenCV4 tutorial/7 Image resample/resize_example.cpp
#include <opencv2/opencv.hpp>

using namespace cv;
int main(void)
{
    Mat src = imread("merge_split.jpg"), dst;
    /* resize 常用模式
            INTER_NEAREST 近邻插值
            INTER_LINEAR 线性插值
            INTER_CUBIC 三次插值
            INTER_AREA 
            INTER_LANCZOS4 三角插值
            INTER_LINEAR_EXACT
            INTER_NEAREST_EXACT
    */
    // 第三个参数为Size()的时候，后面跟的两个浮点数代表图片的宽放缩倍数、高放缩倍数
    resize(src, dst, Size(), 3., 2., INTER_NEAREST);
    imshow("INTER_NEAREST", dst);
    resize(src, dst, Size(), 3., 2., INTER_LINEAR);
    imshow("default", dst);
    waitKey();
    // 输出改变尺寸后的图片(Mat变量)
    imwrite("new_size.png", dst);
}
